lib/blocs/profile/profile_block.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutterapp/blocs/profile/profile_event.dart';
+import 'package:flutterapp/blocs/profile/profile_state.dart';
+import 'package:flutterapp/data/data_load_result.dart';
+import 'package:flutterapp/data/model/user.dart';
+import 'package:flutterapp/data/repository/user_repository.dart';
+
+class ProfileBloc extends Bloc<ProfileEvent, ProfileState> {
+
+  final UserRepositoryAPI userRepository;
+
+  bool _isExpanded = false;
```

Comment:
I'm not sure about isExpanded state. should we have shared states (like user) and ephemeral states (like isExpanded) in blocs? I would rather pass isExpanded as a param with ExpandAchievementsEvent as that's a widget's state

Reply:
When user click on Expand button Bloc must to know current widget state to know what state send to widget.

Reply:
we can pass isExpanded with ExpandAchievementsEvent

Reply:
Let's just rename this to something for Bloc related, rather view related. Like isAchivementsVisible or so

Reply:
Renamed to _isAchievementsExpanded as achievements are always visible (3 achievements are visible by default).